facebook/OpenBIC
# Support system can re-init flash at any time after booting up

Summary:
1. Add a kernel patch to support the re-init-support property for re-initing spi flash at runtime
2. Set the default status of the BIOS SPI selection pin to "L"(connected to PCH)
3. BIC sets the BIOS SPI selection pin to "H" while updating BIOS firmware.
   And then BIC sets to default status after updating successfully.

Test Plan:
1. Build Code: pass
2. Update BIOS firmware: pass
3. Check ME self test result after clearing CMOS: pass

Log:
1. Update BIOS firmware
root@bmc-oob:~# fw-util slot3 --version|grep BIOS
BIOS Version: Y35CLE12
root@bmc-oob:~# fw-util slot3 --update bios /tmp/Y35CLE11.bin
Checking if the server is ready...
Server power status: 1
Powering off the server (cmd 4)...
Server power status: 0
Putting ME into recovery mode...
Switching BIOS SPI MUX for update...
slot_id: 3, comp: 9, intf: 0, img: /tmp/Y35CLE11.bin, force: 0
Init libusb Successful!
...
Updating BIOS, dedup is off, verification is off.
1024 blocks (1024 written, 0 skipped)...finished.
Elapsed time:  265   sec.
Switching BIOS SPI MUX for default value...
Doing ME Reset...
Power-cycling the server...
Upgrade of slot3 : bios succeeded
root@bmc-oob:~# fw-util slot3 --version|grep BIOS
BIOS Version: Y35CLE11
2. Check ME self test result after clearing CMOS
root@bmc-oob:~# bic-util slot3 --clear_cmos
Performing CMOS clear
root@bmc-oob:~# me-util slot3 0x18 0x04
55 00
